scintilla/lexers/LexBatch.cxx

```diff
@@ -284,8 +284,15 @@ void ColouriseBatchDoc(Sci_PositionU startPos, Sci_Position lengthDoc, int initS
 							command = Command::Echo;
 						} else if (StrEqualsAny(s, "do", "else")) {
 							logicalVisibleChars = 0;
-						} else if (StrEqual(s, "goto")) {
-							command = Command::Goto;
+						} else if (StrEqualsAny(s, "exit", "goto")) {
+							if (StrEqual(s, "goto")) {
+								command = Command::Goto;
+							}
+							if (levelCurrent == SC_FOLDLEVELBASE + 1 && parenCount == 0 &&
+								(logicalVisibleChars == 4 || (logicalVisibleChars == 5 &&
+								LexGetPrevChar(sc.currentPos - 4, sc.styler) == '@'))) {

```

Comment:
LexGetPrevChar() can be changed to `styler[styler.GetStartSegment() - 1]`

Reply:
isn't `logicalVisibleChars == 5` means no space between `@` and goto/exit ?

Reply:
https://github.com/zufuliu/notepad2/blob/95b2256ee736153e4c59bce383c4d75686a5742a/scintilla/lexers/LexBatch.cxx#L528-L529
Tested:
![logicalVisibleChars](https://user-images.githubusercontent.com/19585474/124946131-58aa8b80-e041-11eb-8154-b9805c449314.png)